src/platform/test.rs

```diff
@@ -414,6 +414,55 @@ fn receiver_set() {
 }
 
 #[test]
+#[cfg(not(any(feature = "force-inprocess", target_os = "windows", target_os = "android")))]
+fn receiver_set_eintr() {
+    let mut stat: i32 = 0;
+    let (tx0, rx0) = platform::channel().unwrap();
+    let (tx1, rx1) = platform::channel().unwrap();
+    let mut rx_set = OsIpcReceiverSet::new().unwrap();
+    let mut result_set = OsIpcReceiverSet::new().unwrap();
+    rx_set.add(rx1).unwrap();
+    let child_pid = unsafe {
+        fork(|| {
+            // Let the parent know we're ready
+            tx0.send(b"B", vec![], vec![]).unwrap();
+            // Send the result of the select back to the parent
+            match rx_set.select() {
+                Ok(_) => tx0.send(b"\x00", vec![], vec![]).unwrap(),
+                Err(e) => {
+                    let err = format!("{:?}", e);
+                    tx0.send(err.into_bytes().as_slice(), vec![], vec![]).unwrap();
```

Comment:
You can just use `as_bytes()` instead of `into_bytes().as_slice()`.

Note that this is not *exactly* the same, since `into_bytes()` consumes the original string, and the resulting temporary byte string is dropped at the end of the expression; while `as_bytes()` keeps the original string in place. In this case (and many others), this doesn't really make a difference though :-)